### PR TITLE
KIP-416: Notify SourceTask of ACK'd offsets, metadata

### PR DESCRIPTION
**Pull request by ryannedolan**
Added a hook for producer ACKs. See:

https://cwiki.apache.org/confluence/display/KAFKA/KIP-416%3A+Notify+SourceTask+of+ACK%27d+offsets%2C+metadata

### CODE REVIEW

File: connect/api/src/main/java/org/apache/kafka/connect/source/SourceTask.java [hunk 1]
```diff
@@ -88,18 +89,35 @@ public void commit() throws InterruptedException {
...
+     * <p>
+     * Commit an individual {@link SourceRecord} when the callback from the producer client is received.
+     * </p>
+     * <p>
+     * SourceTasks are not required to implement this functionality; Kafka Connect will record offsets
+     * automatically. This hook is provided for systems that also need to store offsets internally
+     * in their own system.
+     * </p>
+     *
+     * @param record {@link SourceRecord} that was successfully sent via the producer
```
Comment (priyank5485) [inline connect/api/src/main/java/org/apache/kafka/connect/source/SourceTask.java:117]: Add @param for recordMetadata

**Comment by ryannedolan** [comment]
retest this please

**Comment by ryannedolan** [comment]
I pulled this change into #6295 and will close this ticket when it is merged.

**Comment by ryannedolan** [comment]
Superseded by PR-6295

**Pull request closed without merging**